src/test/sigencoding_tests.cpp

```diff
@@ -31,9 +31,19 @@ static void CheckSignatureEncodingWithSigHashType(const valtype &vchSig,
 
     std::vector<SigHashType> baseSigHashes;
     for (const BaseSigHashType &baseType : allBaseTypes) {
-        const SigHashType baseSigHash = SigHashType().withBaseType(baseType);
-        baseSigHashes.push_back(baseSigHash);
-        baseSigHashes.push_back(baseSigHash.withAnyoneCanPay(true));
+        const SigHashType sigHashType = SigHashType().withBaseType(baseType);
+        if (hasForkId) {
+            // Check the signatures with the proper forkid flag.
+            baseSigHashes.push_back(sigHashType.withForkId());
+            baseSigHashes.push_back(
+                sigHashType.withForkId().withAnyoneCanPay(true));
+            baseSigHashes.push_back(sigHashType.withBIP341());
+            baseSigHashes.push_back(
+                sigHashType.withBIP341().withAnyoneCanPay(true));
+        } else {
+            baseSigHashes.push_back(sigHashType);
+            baseSigHashes.push_back(sigHashType.withAnyoneCanPay(true));
+        }
```

Comment:
Might want some assertion at the end just to make sure there isn't a extra case being missed.